oceannavigator/frontend/src/components/MainMap.jsx

```diff
@@ -1433,6 +1435,15 @@ const MainMap = forwardRef((props, ref) => {
     }
   };
 
+  const getLineDistance = (line) => {
+    var dist = 0;
+    for (let i = 1; i < line.length; i++) {
+      dist += getDistance(line[i-1],line[i]);
+    }
+      
+    return dist;
+  }
```

Comment:
You're getting a different distance here because your lat and lon values are in the wrong order. You could do something like this to swap them in the array and get the correct distance value:
```suggestion
  const getLineDistance = (line) => {
    var dist = 0;
    for (let i = 1; i < line.length; i++) {
      let start = line[i - 1].reverse()
      let end = line[i].reverse()
      dist += getDistance(start, end);
    }

    return dist;
  }
````

Reply:
Huh, that was one of the first things I tried. I must have messed that up somehow 🤔